llvm/lib/Target/AIE/AIEInterBlockScheduling.h

```diff
@@ -106,14 +107,25 @@ class InterBlockEdges {
 // BlockType determines scheduling priority, direction and safety margin
 // handling.
 enum class BlockType { Regular, Loop, Epilogue };
+enum class SchedulingStage {
+  Scheduling,
+  SchedulingNotConverged,
+  SchedulingDone,
+  Pipelining,
+  PipeliningDone,
+  PipeliningFailed
+};
```

Comment:
Maybe give more details about the states and the legal transitions? In particular:
- Is `SchedulingNotConverged` an error state, or do we still have a way to schedule conservatively? In this case, is it equivalent to `SchedulingDone`?
- Similar questions for `PipeliningFailed`
- Is `SchedulingDone` equivalent to `Pipelining`, or is `SchedulingDone` a final state?